### PR TITLE
Add Hailo export and inference for the YOLO26 depth task

### PR DESCRIPTION
**Pull request by JESUSROYETH**
`format="hailo"` already covers detection, segmentation, pose, OBB, classification and semantic segmentation, but not depth. So a user who trains a YOLO26 depth model has no way to run it on a Hailo chip today. Depth is a good fit for the 8L: it is one dense head on the same YOLO26 backbone we already export for detection, and it runs well on the real chip.

I validated the whole path on a physical Hailo-8L (Hailo-8L PCIe, DFC 3.33, HailoRT 4.23) before opening this:

- `yolo26{n,s,m}-depth` compile to a single HEF on `hailo8l`, no split — the C2PSA `matmul` stays in one context because the depth head is a single dense output.
- On the chip `yolo26n-depth` runs at 46 FPS (21.6 ms/frame, infer only).
- Depth-map fidelity vs the PyTorch model, measured on the chip: Pearson r ≈ 0.95, δ<1.25 ≈ 0.90 (per-image median-aligned, in-domain calibration).

## Changes

Two source files plus docs, and `head.py` is not touched. The design mirrors the semantic task that was just added to the same Hailo path.

`ultralytics/engine/exporter.py` routes depth through `export_hailo`:

- The guard accepts `depth` (YOLO26 only, like semantic).
- The end node is the raw logit — the head's final `Conv` (`head.3`), before the `clamp/exp`, calibration and 4x upsample. Same string convention detect already uses for its `.2/Conv`, so no new head attribute.
- The output goes out in `a16`, reusing the one2one path (no NMS, no activation change). a8 collapses the depth map, a16 keeps the range. This is the config I validated on the chip.
- `metadata.yaml` carries `cal_a`/`cal_b`, the head's learned log-affine calibration buffers, because the decode runs on the host.

`ultralytics/nn/backends/hailo.py` decodes the depth output:

- The task guard accepts `depth`, and `forward` dispatches to a new `_decode_depth`.
- `_decode_depth` mirrors `Depth.forward` from the logit onward: `clamp(-4,5).exp()` and the `pow(cal_a)·exp(cal_b)` calibration. It returns the map at head resolution `(B, 1, H/4, W/4)` and lets `DepthPredictor.postprocess` resize it with `scale_masks`, the same path the PyTorch model takes at inference — so the Hailo output matches `yolo predict` exactly (max |host − reference| = 0.0 over the test images). This is the same low-res-then-resize the semantic branch does one step above.

Docs: the depth task is added everywhere the Hailo task set is enumerated, the same way the semantic PR kept them in sync — the integration page (task table, hardware table, mechanism note, limitations, calibration section, FAQ, summary), the integrations index, and the model-deployment-options guide. The calibration section also gets a one-line note that the dense depth head is sensitive to the calibration domain, since that is the main problem I found while validating.

## Reused / consolidated

The generic scaffold already routes depth end to end, so this stays small:

- `Depth` is already in the export-mode tuple in `exporter.py`, so the ONNX graph exports correctly with nothing to add.
- `TASK2CALIBRATIONDATA["depth"]` already exists (`depth8.yaml`), so the calibration default is depth data, not `coco128`.
- `apply_metadata` setattrs every metadata key, so `cal_a`/`cal_b` need no new inference plumbing.
- The `a16` model-script branch is shared with the one2one detect path, not a separate script only for depth.
- `DepthPredictor.postprocess` consumes the decoded map as is and resizes it with `scale_masks` — no predictor or autobackend change.
- `nms=False` is already forced for depth before export, so no NMS path is reachable.

## Deleted

Deleted: nothing. This is new capability, a task the Hailo path did not support before, so there is no old code to remove. The two edits that could look like additions are replacements: the semantic-only YOLO26 guard becomes a `{semantic, depth}` guard, and the `if one2one` model-script branch becomes `if one2one or depth`.

## Validation

Measured on a physical Hailo-8L (DFC 3.33 / HailoRT 4.23): n/s/m compile to a single HEF, `yolo26n-depth` at 46 FPS, depth-map Pearson r ≈ 0.95 / δ<1.25 ≈ 0.90 vs PyTorch (median-aligned, in-domain calibration), host decode bit-exact.

```bash
# export (needs the Hailo DFC installed)
yolo export model=yolo26n-depth.pt format=hailo name=hailo8l

# run on the chip
yolo predict task=depth model=yolo26n-depth_hailo_model source='https://ultralytics.com/images/bus.jpg'
```

One important note, I hit this myself while validating: the dense depth head is more sensitive to the calibration set than detection. Calibrating with a detection set like `coco128` collapses the logit and the map goes flat (r ≈ 0.4). In-domain images recover it, and the amount matters — 256 images get r ≈ 0.92, 1024 get r ≈ 0.95. The export default `depth8.yaml` covers the common case, but it is a minimum. Users who care about accuracy should point `data=` at a larger in-domain set.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds Hailo export and inference support for YOLO26 depth estimation models, including host-side decoding of metric depth maps.

### 📊 Key Changes
- Extends Hailo export validation, documentation, compatibility tables, and calibration guidance for YOLO26 depth models.
- Compiles the depth head’s final dense logit convolution with `a16` precision and stores calibration metadata in the HEF metadata.
- Adds Hailo backend depth decoding that mirrors the YOLO26 `Depth` head’s clamp, exponential transform, and learned log-affine calibration.
- Documents Hailo hardware support and the host-side depth reconstruction workflow.

### 🎯 Purpose & Impact
- Enables deployment of YOLO26 depth models on Hailo-8/8L and compatible Hailo targets.
- Preserves metric depth reconstruction while keeping post-processing outside the HEF for compatibility and fidelity.
- Improves deployment guidance with task-specific calibration recommendations, while noting that validation beyond Hailo-8L remains limited.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JESUSROYETH, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message helps with the initial review; an Ultralytics engineer will assist as needed. Please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean PR. The depth export path is consistently wired through Hailo format validation, ONNX endpoint selection, output quantization, metadata persistence, and host-side reconstruction. The host decode correctly mirrors `Depth.forward` before export-only upsampling, allowing the existing depth predictor postprocessing path to resize and unletterbox results.

**Comment by glenn-jocher** [comment]
Thanks for the detailed hardware validation. CI passes, but patch coverage is only 8.3%; please add focused unit tests for `_decode_depth()`—including default and `cal_a`/`cal_b` metadata—and the Hailo depth routing/metadata path, mocking DFC/HailoRT so tests run without hardware.

**Comment by glenn-jocher** [comment]
Reviewed deeply before merge — this is the right minimal shape for a new-task backend extension: depth slots into every existing seam (task guard sets, end-node chain, shared a16 model-script branch, metadata dict, `_decode_*` dispatch) with no parallel plumbing; +31/−8 source lines excluding tests/docs. Verified empirically: exported `DepthModel('yolo26n-depth.yaml')` through the real `torch2onnx` path and confirmed the `/model.23/head/head.3/Conv` end node exists with exactly `Clip → Exp → Pow → Exp_1 → Mul → Resize` after the cut — matching what `_decode_depth` reproduces host-side, with the dropped 4x Resize correctly delegated to `scale_masks` in the predictor, so Hailo follows the identical resize path as native predict. All four mocked tests pass locally, bit-exactness holds for identity and learned calibration, and the Hailo-8L numbers (46 FPS, r≈0.95) close the hardware loop.

**Comment by glenn-jocher** [comment]
Slimmed the diff: dropped the mocked test additions — the ONNX end-node verification and Hailo-8L hardware validation above are the record. Source diff is +31/−8.
